app/src/main/java/com/celzero/bravedns/service/BraveVPNService.kt

```diff
@@ -768,21 +823,22 @@ class BraveVPNService : VpnService(), ConnectionMonitor.NetworkListener, Protect
             // instead call spawnServerUpdate()
             // spawnServerUpdate() - Will overwrite the tunnel values with new values.
             if (connectionMonitor != null) {
-                spawnServerUpdate()
+                CoroutineScope(Dispatchers.IO).launch {
+                    spawnServerUpdate(appMode.makeTunnelDataClass())
+                }
                 return Service.START_REDELIVER_INTENT
             }
 
             // Register for ConnectivityManager
             // FIXME #200 - Move it to Koin
             connectionMonitor = ConnectionMonitor(this, this)
-
-            restartVpn(appMode.getDNSMode(), appMode.getFirewallMode(), appMode.getProxyMode())
-
+            CoroutineScope(Dispatchers.IO).launch {
```

Comment:
The `appMode.newTunnelMode` is made as suspend function as it can access database while creating the object.

Reply:
Then make `restartVpn` a `suspend` function? It may not need a `new Thread`. Also, GoVpnAdapter is a java file, so check documentation once on how that transition may happen.

Reply:
done.